File: CS/CS.NET/Windows Forms .NET/FormToForm/EventArgs/CS2008FormToForm/CS2008FormToForm/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;


public partial class Form1 : Form
{
    public Form1()
    {
        InitializeComponent();
    }

    private Form2 F2;

    private event CustomEventHandler Form1TextBoxEvent;

    private void Form2_Form2TextBoxText(object sender, CustomEventArgs e)
    {
        textBox1.Text  = e.TextBoxTextChanged;
    }

    private void OnForm1TextBoxEvent(CustomEventArgs e)
    {
        if(Form1TextBoxEvent != null)
            Form1TextBoxEvent(this, e);
    }
        
    private void button1_Click(object sender, EventArgs e)
    {
        F2 = new Form2();

        // Form1TextBoxEvent += new CustomEventHandler(F2.Form1_Form1TextBoxEvent);
        Form1TextBoxEvent += F2.Form1_Form1TextBoxEvent;

        CustomEventArgs Form1TextBoxEventArgs = new CustomEventArgs(textBox1.Text);
        OnForm1TextBoxEvent(Form1TextBoxEventArgs);

        // F2.Form2TextBoxEvent += new CustomEventHandler(Form2_Form2TextBoxText);
        F2.Form2TextBoxEvent += Form2_Form2TextBoxText;

        F2.ShowInTaskbar = false;
        F2.ShowDialog();
    }
}

File: CS/CS.NET-VB.NET/FormToForm/VB2008FormToForm/VB2008FormToForm/Form1.vb
Public Class Form1

    Private F2 As Form2

    ' internal event CustomEventHandler Form1TextBoxEvent;
    Private Event Form1TextBoxEvent As CustomEventHandler


    Private Sub Form2_Form2TextBoxText(ByVal sender As System.Object, ByVal e As CustomEventArgs)

        TextBox1.Text = e.TextBoxTextChanged

    End Sub

    Private Sub OnForm1TextBoxEvent(ByVal e As CustomEventArgs)

        Try
            RaiseEvent Form1TextBoxEvent(Me, e)
        Catch ex As Exception
            MessageBox.Show(ex.ToString())
        End Try

    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click

        F2 = New Form2()

        ' Form1TextBoxEvent += F2.Form1_Form1TextBoxEvent;
        AddHandler Form1TextBoxEvent, AddressOf F2.Form1_Form1TextBoxEvent

        Dim Form1TextBoxEventArgs As New CustomEventArgs(TextBox1.Text)
        OnForm1TextBoxEvent(Form1TextBoxEventArgs)

        ' F2.Form2TextBoxEvent += Form2_Form2TextBoxText;
        AddHandler F2.Form2TextBoxEvent, AddressOf Form2_Form2TextBoxText

        F2.ShowInTaskbar = False
        F2.ShowDialog()

    End Sub

End Class
